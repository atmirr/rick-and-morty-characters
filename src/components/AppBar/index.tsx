// @flow
import React from 'react';
import styled from 'styled-components';
import RickAndMortyLogo from '../../assets/svgs/rick-and-morty-logo.svg';

const HEADER_BOX_SHADOW =
  '0px 2px 4px -1px rgba(0, 0, 0, 0.2), 0px 4px 5px 0px rgba(0, 0, 0, 0.14), 0px 1px 10px 0px rgba(0, 0, 0, 0.12)';

const Header = styled.div`
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  box-sizing: border-box;
  background-color: #11b0c8;
  border-bottom: 1px solid #06a3bb;
  padding: 10px 25px;
  height: 90px;
  box-shadow: ${HEADER_BOX_SHADOW};
`;

const Image = styled.img`
  height: 60px;
`;

function AppBar(): React.ReactElement {
  return (
    <Header>
      <Image src={RickAndMortyLogo} alt="Rick & Morty logo" />
    </Header>
  );
}

export default AppBar;
