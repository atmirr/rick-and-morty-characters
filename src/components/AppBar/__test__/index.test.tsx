import React from 'react';
import 'jest-styled-components';
import '@testing-library/jest-dom';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import AppBar from '../index';

describe('<AppBar />', () => {
  it('should renders the component', () => {
    const { container } = render(<AppBar />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should display the logo', () => {
    const { container } = render(<AppBar />);
    const imageNodes = container.querySelectorAll('img');
    expect(imageNodes.length).toBe(1);
  });
});
