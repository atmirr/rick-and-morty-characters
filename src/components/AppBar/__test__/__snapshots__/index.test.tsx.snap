// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppBar /> should renders the component 1`] = `
.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  box-sizing: border-box;
  background-color: #11b0c8;
  border-bottom: 1px solid #06a3bb;
  padding: 10px 25px;
  height: 90px;
  box-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);
}

.c1 {
  height: 60px;
}

<div
  class="c0"
>
  <img
    alt="Rick & Morty logo"
    class="c1"
    src="rick-and-morty-logo.svg"
  />
</div>
`;
